src/reflection.cpp

```diff
@@ -481,4 +481,232 @@ Offset<const Table *> CopyTable(FlatBufferBuilder &fbb,
   }
 }
 
+bool VerifyStruct(flatbuffers::Verifier &v,
+                  const flatbuffers::Table &parentTable,
+                  voffset_t fieldOffset,
+                  const reflection::Object &obj,
+                  bool isRequired) {
+  assert(obj.is_struct());
+  auto offset = parentTable.GetOptionalFieldOffset(fieldOffset);
+  if (!isRequired) {
```

Comment:
Maybe for structs it is not ideal, but I generally see repetitive code here that I would like to see better factored. The simplest part is that `v.Verify(reinterpret_cast<const uint8_t*>(&parentTable) + offset, obj.bytesize());` appears twice, which I don't think is necessary, and the second part is checking for `isRequired` which I think at least for some fields can be done more centrally.